tools/expotools/src/commands/UpdateReactNativeDocs.ts

```diff
@@ -0,0 +1,252 @@
+  }
+
+  try {
+    await spawnAsync('git', ['checkout', options.to], { cwd: RN_WEBSITE_DIR });
+  } catch (error) {
+    throw new Error(`The --to commit "${options.to}" doesn't exists in the submodule.`);
+  }
+}
+
+async function getLocalFilesAsync(options: Options) {
+  console.log(`Resolving local docs from ${chalk.yellow(options.sdk)} folder...`);
+
+  const versionedDocsPath = path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native');
+  const files = await fs.promises.readdir(versionedDocsPath);
+
+  return files.map(entry => entry.replace('.md', ''));
+}
+
+async function getUpstreamFilesAsync(options: Options) {
+  console.log(`Resolving upstream docs from ${chalk.cyan('react-native-website')} submodule...`);
+
+  const docsPath = path.join(RN_WEBSITE_DIR, 'docs');
+  const sidebarPath = path.join(RN_WEBSITE_DIR, 'website', 'sidebars.json');
+  const sidebarData = await fs.readJson(sidebarPath);
+
+  let relevantNestedDocs: any[] = [];
+  try {
+    relevantNestedDocs = [
+      ...sidebarData.api.APIs,
+      ...sidebarData.components['Core Components'],
+      ...sidebarData.components.Props,
+    ];
+  } catch (error) {
+    console.log('');
+    console.log(`There was an ${chalk.red('error')} extracting the sidebar information.`);
+    console.log(`Please double-check the sidebar and update the relevant sections in this script.`);
+    console.log(`${chalk.dim(`- ${sidebarPath}`)}`);
+    console.log('');
+    throw error;
+  }
+
+  const upstreamDocs: any[] = [];
+  const relevantDocs: any = relevantNestedDocs.map(entry => {
+    if (typeof entry === 'object' && Array.isArray(entry.ids)) {
+      return entry.ids;
+    }
+
+    if (typeof entry === 'string') {
+      return entry;
+    }
+  });
+
+  for (const entry of relevantDocs.flat()) {
+    const docExists = await fs.pathExists(path.join(docsPath, `${entry}.md`));
+    const docIsIgnored = DOCS_IGNORED.includes(entry);
+
+    if (docExists && !docIsIgnored) {
+      upstreamDocs.push(entry);
+    }
+  }
+
+  return upstreamDocs;
+}
+
+function getDocsSummary(localFiles: string[], upstreamFiles: string[]): DocsSummary {
+  const removed = localFiles.filter(entry => !upstreamFiles.includes(entry));
+  const added = upstreamFiles.filter(entry => !localFiles.includes(entry));
+
+  const changed = upstreamFiles.filter(entry => (
+    !(removed.includes(entry) || added.includes(entry))
+  ));
+
+  return { removed, added, changed };
+}
+
+async function applyRemovedFilesAsync(options: Options, summary: DocsSummary) {
+  if (!summary.removed.length) {
+    return console.log(`Upstream did not ${chalk.red('remove')} any files`);
+  }
+
+  for (const entry of summary.removed) {
+    if (entry.startsWith(PREFIX_REMOVED)) {
+      continue;
+    }
+
+    await fs.move(
+      path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native', `${entry}.md`),
+      path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native', `${PREFIX_REMOVED}${entry}.md`),
```

Comment:
I think it's worth to join this directory once and then join with just filenames — easier to read, easier to maintain